python/storage/kserve_storage/kserve_storage.py

```diff
@@ -55,6 +56,9 @@
 _HDFS_SECRET_DIRECTORY = "/var/secrets/kserve-hdfscreds"
 _HDFS_FILE_SECRETS = ["KERBEROS_KEYTAB", "TLS_CERT", "TLS_KEY", "TLS_CA"]
 
+# Azure async download configuration
+_AZURE_MAX_FILE_CONCURRENCY = int(os.getenv("AZURE_MAX_FILE_CONCURRENCY", "4"))
```

Comment:
Does it behave well in a very limited namespace? Let's say 500mi of CPU and 128MB of memory.